Userland/Utilities/touch.cpp

```diff
@@ -1,51 +1,253 @@
 /*
  * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Ariel Don <ariel@arieldon.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/CheckedFormatString.h>
+#include <AK/GenericLexer.h>
+#include <AK/Time.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
+#include <LibTimeZone/TimeZone.h>
+#include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <time.h>
 
-static bool file_exists(String const& path)
+static String program_name;
+
+template<typename... Parameters>
+[[noreturn]] static void err(CheckedFormatString<Parameters...>&& fmtstr, Parameters const&... parameters)
+{
+    warn("{}: ", program_name);
+    warnln(move(fmtstr), parameters...);
+    exit(1);
+}
+
+inline static bool validate_timestamp(unsigned year, unsigned month, unsigned day, unsigned hour, unsigned minute, unsigned second)
 {
-    struct stat st;
-    int rc = stat(path.characters(), &st);
-    if (rc < 0) {
-        if (errno == ENOENT)
-            return false;
+    return (year >= 1970) && (month >= 1 && month <= 12) && (day >= 1 && day <= (unsigned)days_in_month(year, month)) && (hour <= 23) && (minute <= 59) && (second <= 59);
```

Comment:
no c-style cast please

Reply:
I converted it to a static cast. I also checked the other changes, and I didn't find any other casts.